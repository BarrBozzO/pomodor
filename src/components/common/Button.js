import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const StyledButton = styled.button`
  box-sizing: border-box;
  border: 1px solid #f44336;
  border-radius: 20px;
  background: ${({ primary }) => (primary ? "#f44336" : "#ffffff")};
  font-size: ${({ small }) => (small ? "14px" : "16px")};
  line-height: normal;
  padding: ${({ small }) => (small ? "8px 14px" : "10px 16px")};
  margin: 0;
  color: ${({ primary }) => (primary ? "#ffffff" : "#f44336")};
  cursor: pointer;
  font-weight: 600;

  & ~ & {
    margin-left: 10px;
  }

  &:disabled {
    opacity: 0.6;
  }
  @media (max-width: 360px) {
    font-size: 14px;
    padding: 10px 12px;
  }
`;

function Button(props) {
  return (
    <StyledButton
      onClick={props.handleClick}
      disabled={props.disabled}
      primary={props.primary}
      small={props.small}
      className={props.className}
      type={props.type}
    >
      {props.value}
    </StyledButton>
  );
}

Button.propTypes = {
  handleClick: PropTypes.func,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  disabled: PropTypes.bool,
  primary: PropTypes.bool,
  small: PropTypes.bool,
  type: PropTypes.string,
  className: PropTypes.string
};

Button.defaultProps = {
  disabled: false,
  primary: false,
  small: false,
  type: "button"
};

export default Button;
