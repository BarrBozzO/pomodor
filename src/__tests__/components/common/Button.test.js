import React from "react";
import renderer from "react-test-renderer";

import Button from "../../../components/common/Button";

it("should render Button component correctly", () => {
  const handleClick = jest.fn(),
    value = "testBtn",
    disabled = false,
    primary = true,
    type = "submit";

  const tree = renderer
    .create(
      <Button
        value={value}
        disabled={disabled}
        primary={primary}
        handleClick={handleClick}
        type={type}
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
