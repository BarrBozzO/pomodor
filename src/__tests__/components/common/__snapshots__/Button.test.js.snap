// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Button component correctly 1`] = `
<button
  className="sc-bdVaJa bUPfJD"
  disabled={false}
  onClick={[MockFunction]}
>
  testBtn
</button>
`;
